### PR TITLE
Rename web-view to webview for consistency

### PR DESCRIPTION
**Pull request by zeke**
The tag itself is `webview`, but there are several places in the code that refer to it as `web-view`. This leads to confusion, or at least a slight cognitive delay when working with them or looking up their documentation. This PR attempts to normalize all that, so `webview` will always be the thing you type in the Electron codebase, in the docs, and in practical use.

Also created https://github.com/electron/electron.atom.io/pull/562 so the website will continue to properly redirect requests to the old `/docs/api/web-view-tag` URL.

Let a few other PRs land first:

- [x] https://github.com/electron/electron/pull/7852
- [x] https://github.com/electron/electron/pull/7863

### CODE REVIEW

**Comment by kevinsawicki** [comment]
Sounds good to me but would prefer to land to #7852 and #7863 first.

**Comment by kevinsawicki** [comment]
Both PRs have landed now, let me 👍

**Comment by zeke** [comment]
Rebased. This should be ready to go again, assuming CI passes.

File: lib/renderer/webview/webview-attributes.js [hunk 1]
```diff
@@ -1,8 +1,8 @@
 'use strict'
 
-const WebViewImpl = require('./web-view')
+const WebViewImpl = require('./webview')
 const guestViewInternal = require('./guest-view-internal')
-const webViewConstants = require('./web-view-constants')
+const webViewConstants = require('./webview-constants')
```
Comment (kevinsawicki) [inline lib/renderer/webview/webview-attributes.js]: After thinking about this more and reviewing the diff again, I'm not so sure about these renames, the constructor for this object is `new WebView`, not `new Webview` so it feels right to have it dashed in the file name(s), similar to `browser-window.js`.

Comment (zeke) [inline lib/renderer/webview/webview-attributes.js]: > the constructor for this object is new WebView

Great point. I only searched for dasherized variants.

**Comment by zeke** [comment]
Okay I updated this to only rename stuff related to the `webview` tag.

**Comment by zeke** [comment]
Rebased again to bring in changes to webview docs in other PRs. Ready to 🚢

**Comment by MarshallOfSound** [comment]
@zeke Still conflicting 😢

**Comment by zeke** [comment]
Re-re-rebased. ⚡️

**Comment by zeke** [comment]
@kevinsawicki are you still 👍 on this?

(The two CI failures are from timeouts)

**Comment by kevinsawicki** [comment]
It looks like `spec/fixtures/pages/web-view-log-process.html` was renamed but no test was updated with its new path. If it is unused then I'd say just delete it in this pull request completely.

**Comment by zeke** [comment]
> It looks like spec/fixtures/pages/web-view-log-process.html was renamed

Good call. Yeah I couldn't find an instance of it in the tests but figured maybe there was some metaprogramming going on. I will try removing it.

**Comment by zeke** [comment]
I had to bump the webview spec timeout to 3 seconds to get the Travis Node 4 Linux x64 build to pass.

**Pull request merged**